akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -151,7 +151,14 @@ import akka.stream.stage._
 
   }
 
-  final class IdleTimeoutBidi[I, O](val timeout: FiniteDuration) extends GraphStage[BidiShape[I, I, O, O]] {
+  private object IdleTimeoutBidi {
+    val defaultCreateFailure: FiniteDuration => Throwable = timeout =>
+      new StreamIdleTimeoutException(s"No elements passed in the last ${timeout.toCoarsest}.")
+  }
+  final class IdleTimeoutBidi[I, O](
+      val timeout: FiniteDuration,
+      createFailure: FiniteDuration => Throwable = IdleTimeoutBidi.defaultCreateFailure)
```

Comment:
Add a overloaded method instead of default parameter? And I think `failureCreator`  maybe a better name.

Reply:
default param is fine here since constructor is part of internal API